cpp/tests/utilities/column_wrapper.hpp

```diff
@@ -109,13 +111,68 @@ class column_wrapper {
  *`[begin,end)`
  **/
 template <typename Element, typename InputIterator>
-rmm::device_buffer make_elements(InputIterator begin, InputIterator end)
+rmm::device_buffer make_elements(
+  InputIterator begin,
+  InputIterator end,
+  typename std::enable_if<
+    std::is_convertible<typename std::iterator_traits<InputIterator>::value_type,
+                        Element>::value>::type* = nullptr)
 {
   static_assert(cudf::is_fixed_width<Element>(), "Unexpected non-fixed width type.");
   thrust::host_vector<Element> elements(begin, end);
   return rmm::device_buffer{elements.data(), elements.size() * sizeof(Element)};
 }
 
+// This type trait can be specialized for different `From` and/or `To` types, or the conditional
+// within it can be nested for future types. This is used in the context of `make_elements`
+// when `Element` cannot be implicitly constructed from the items returned by the `InputIterator`.
+template <typename From, typename To>
+struct argument_type_for_explicit_construction {
+  using argument_type = typename std::conditional<
+    cudf::is_duration_t<To>::value,
+    typename std::
+      conditional<std::is_convertible<From, typename To::rep>::value, typename To::rep, void>::type,
+    From>::type;
+};
+
+/**
+ * @brief Creates a `device_buffer` containing the elements in the range
+ * `[begin,end)`.
+ *
+ * This is applicable for types that do not allow implicit construction of `Element` from the
+ * arithmetic items returned by the `InputIterator`. The items returned by the iterator
+ * may have to be casted to integral types if the `Element` is a duration type. This behavior
+ * can be specialized thusly.
+ * The `argument_type_for_explicit_construction` type trait can either be specialized
```

Comment:
Can you explain the cases that came up where this was necessary? Overriding explicit construction can lead to very surprising behavior.

Reply:
i have made the change for the duration and the timestamp cast tests to work with integral types.